File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/animation/UXAnimationTimer.php
<?php
namespace php\gui\animation;

/**
 * Class UXAnimationTimer
 * @package php\gui\animation
 *
 * @packages gui, javafx
 */
class UXAnimationTimer
{
    const FRAME_INTERVAL = 1 / 60;
    const FRAME_INTERVAL_MS = self::FRAME_INTERVAL * 1000;

    /**
     * @param callable $handler ($now)
     */
    public function __construct(callable $handler)
    {
    }

    public function start()
    {
    }

    public function stop()
    {
    }
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/animation/UXAnimationTimer.java
package org.develnext.jphp.ext.javafx.classes.animation;

import javafx.animation.AnimationTimer;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.invoke.Invoker;
import php.runtime.lang.BaseWrapper;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "animation\\UXAnimationTimer")
public class UXAnimationTimer extends BaseWrapper<AnimationTimer> {
    public static final float FRAME_INTERVAL = 1 / 60f;
    public static final float FRAME_INTERVAL_MS = FRAME_INTERVAL * 1000;

    interface WrappedInterface {
        void start();
        void stop();
    }

    protected Invoker handler;

    class AnimationTimerImpl extends AnimationTimer {
        @Override
        public void handle(long now) {
            if (handler.callAny(now).toBoolean()) {
                stop();
            }
        }
    }

    public UXAnimationTimer(Environment env, AnimationTimer wrappedObject) {
        super(env, wrappedObject);
    }

    public UXAnimationTimer(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct(Invoker invoker) {
        handler = invoker;
        __wrappedObject = new AnimationTimerImpl();
    }
}
